fcrepo-clients/src/main/java/edu/unc/lib/dl/fedora/ManagementClient.java

```diff
@@ -339,6 +340,28 @@ public Datastream getDatastream(PID pid, String dsID, String asOfDateTime) throw
 		GetDatastreamResponse resp = (GetDatastreamResponse) this.callService(req, Action.getDatastream);
 		return resp.getDatastream();
 	}
+	
+	public Boolean dataStreamExists(PID pid, String dsID) {
+		UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(this.getFedoraContextUrl());
+		builder.pathSegment("objects");
+		builder.pathSegment(pid.getPid());
+		builder.pathSegment("datastreams");
+		builder.pathSegment(dsID);
+
+		HttpHead method = new HttpHead(builder.build().encode().toUriString());
+
+		try (CloseableHttpResponse httpResp = httpClient.execute(method)) {
+			int statusCode = httpResp.getStatusLine().getStatusCode();
+
+			if (statusCode == 200) {
+				return true;
+			}
+
+			return false;
+		} catch (Exception e) {
```

Comment:
Added logging on error. The most common we got was 500 for a malformed request to fedora.